### PR TITLE
[hlm-707]:  Added Endpoint to search based on syncId, status, filesto…

### PR DESCRIPTION
**Pull request by jitendrac-egov**
…reid, and referenceId - jitendra

### CODE REVIEW

File: digit-health-services/health-sync-service/src/main/java/org/digit/health/sync/repository/SyncRepository.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import org.digit.health.sync.web.models.request.SyncSearchDto;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.core.BeanPropertyRowMapper;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.stereotype.Repository;
+
+import java.util.List;
+
+@Repository
+public class SyncRepository  {
```
Comment (jayantp-egov) [inline digit-health-services/health-sync-service/src/main/java/org/digit/health/sync/repository/SyncRepository.java:13]: Make SyncLogRepository an interface and create an implementation called DefaultSyncLogRepository

File: digit-health-services/health-sync-service/src/main/java/org/digit/health/sync/web/models/dao/SyncData.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@Builder
+public class SyncData {
```
Comment (jayantp-egov) [inline digit-health-services/health-sync-service/src/main/java/org/digit/health/sync/web/models/dao/SyncData.java:14]: Make this SyncLogData

File: digit-health-services/health-sync-service/src/test/java/org/digit/health/sync/repository/SyncRepositoryTest.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    @InjectMocks
+    private SyncRepository syncRepository;
+
+    @BeforeEach
+    void setUp() {
+        syncRepository = new SyncRepository(jdbcTemplate,syncQueryBuilder);
+    }
+
+    @Test
+    @DisplayName("should successfully get results from sync repository")
```
Comment (jayantp-egov) [inline digit-health-services/health-sync-service/src/test/java/org/digit/health/sync/repository/SyncRepositoryTest.java:48]: Make this test proper

File: digit-health-services/health-sync-service/src/test/java/org/digit/health/sync/service/FileSyncServiceTest.java [hunk 4]
```diff
@@ -125,6 +136,21 @@ void shouldThrowCustomExceptionWhenFileIsNotJson() throws IOException {
         assertThatThrownBy(() -> fileSyncService.syncUp(syncUpDto)).isInstanceOf(CustomException.class);
     }
 
+    @Test
+    @DisplayName("should successfully get results from sync repository")
```
Comment (jayantp-egov) [inline digit-health-services/health-sync-service/src/test/java/org/digit/health/sync/service/FileSyncServiceTest.java:140]: Make this test proper

File: digit-health-services/health-sync-service/src/main/java/org/digit/health/sync/repository/SyncQueryBuilder.java [hunk 5]
```diff
@@ -0,0 +1,34 @@
+package org.digit.health.sync.repository;
+
+import lombok.extern.slf4j.Slf4j;
+import org.digit.health.sync.web.models.request.SyncSearchDto;
+import org.springframework.stereotype.Component;
+
+@Slf4j
+@Component
+public class SyncQueryBuilder {
```
Comment (jayantp-egov) [inline digit-health-services/health-sync-service/src/main/java/org/digit/health/sync/repository/SyncQueryBuilder.java:9]: Make an interface SyncLogQueryBuilder and implement with DefaultSyncLogQueryBuilder

File: digit-health-services/health-sync-service/src/main/java/org/digit/health/sync/web/models/request/SyncSearchDto.java [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import org.digit.health.sync.web.models.ReferenceId;
+import org.egov.common.contract.request.RequestInfo;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@Builder
+public class SyncSearchDto {
```
Comment (jayantp-egov) [inline digit-health-services/health-sync-service/src/main/java/org/digit/health/sync/web/models/request/SyncSearchDto.java:14]: SyncLogSearchDto

File: digit-health-services/health-sync-service/src/main/java/org/digit/health/sync/web/models/request/SyncSearchMapper.java [hunk 7]
```diff
@@ -0,0 +1,14 @@
+package org.digit.health.sync.web.models.request;
+
+import org.mapstruct.Mapper;
+import org.mapstruct.factory.Mappers;
+
+@Mapper
+public interface SyncSearchMapper {
```
Comment (jayantp-egov) [inline digit-health-services/health-sync-service/src/main/java/org/digit/health/sync/web/models/request/SyncSearchMapper.java:7]: SyncLogSearchMapper

File: digit-health-services/health-sync-service/src/main/java/org/digit/health/sync/web/models/request/SyncSearchRequest.java [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+import org.egov.common.contract.request.RequestInfo;
+import org.springframework.beans.factory.annotation.Required;
+
+import javax.validation.constraints.NotNull;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@Builder
+public class SyncSearchRequest {
```
Comment (jayantp-egov) [inline digit-health-services/health-sync-service/src/main/java/org/digit/health/sync/web/models/request/SyncSearchRequest.java:18]: SyncLogSearchRequest

**Review by jayantp-egov — CHANGES_REQUESTED** [review]
Make sure you update respective Test class names and method names as per refactors.

File: digit-health-services/health-sync-service/src/main/java/org/digit/health/sync/repository/DefaultSyncLogRepository.java [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+import java.util.Map;
+
+@Repository("defaultSyncLogRepository")
+public class DefaultSyncLogRepository implements SyncLogRepository{
+
+    private NamedParameterJdbcTemplate namedParameterJdbcTemplate;
+
+    private final SyncLogQueryBuilder syncLogQueryBuilder;
+
+    @Autowired
```
Comment (jayantp-egov) [inline digit-health-services/health-sync-service/src/main/java/org/digit/health/sync/repository/DefaultSyncLogRepository.java:21]: Line length should not be more than 100-120 chars

File: digit-health-services/health-sync-service/src/main/java/org/digit/health/sync/repository/DefaultSyncLogRepository.java [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.stereotype.Repository;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@Repository("defaultSyncLogRepository")
+public class DefaultSyncLogRepository implements SyncLogRepository{
+
+    private NamedParameterJdbcTemplate namedParameterJdbcTemplate;
```
Comment (jayantp-egov) [inline digit-health-services/health-sync-service/src/main/java/org/digit/health/sync/repository/DefaultSyncLogRepository.java]: Make this final

File: digit-health-services/health-sync-service/src/test/java/org/digit/health/sync/repository/DefaultSyncLogRepositoryTest.java [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+
+import static org.junit.Assert.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+class DefaultSyncLogRepositoryTest {
+
+    @Mock
+    NamedParameterJdbcTemplate jdbcTemplate;
```
Comment (jayantp-egov) [inline digit-health-services/health-sync-service/src/test/java/org/digit/health/sync/repository/DefaultSyncLogRepositoryTest.java:31]: Make it private

File: digit-health-services/health-sync-service/src/test/java/org/digit/health/sync/repository/DefaultSyncLogRepositoryTest.java [hunk 12]
```diff
@@ -0,0 +1,66 @@
...
+
+    @Mock
+    @Qualifier("defaultSyncLogQueryBuilder")
+    SyncLogQueryBuilder syncLogQueryBuilder;
```
Comment (jayantp-egov) [inline digit-health-services/health-sync-service/src/test/java/org/digit/health/sync/repository/DefaultSyncLogRepositoryTest.java:35]: Make it private

**Review by jayantp-egov — CHANGES_REQUESTED** [review]

File: digit-health-services/health-sync-service/src/test/java/org/digit/health/sync/repository/DefaultSyncLogRepositoryTest.java (hunk 11 above)
Comment (jitendrac-egov) [reply digit-health-services/health-sync-service/src/test/java/org/digit/health/sync/repository/DefaultSyncLogRepositoryTest.java]: done

File: digit-health-services/health-sync-service/src/test/java/org/digit/health/sync/repository/DefaultSyncLogRepositoryTest.java (hunk 12 above)
Comment (jitendrac-egov) [reply digit-health-services/health-sync-service/src/test/java/org/digit/health/sync/repository/DefaultSyncLogRepositoryTest.java]: resolved

File: digit-health-services/health-sync-service/src/test/java/org/digit/health/sync/service/FileSyncServiceTest.java (hunk 4 above)
Comment (jitendrac-egov) [reply digit-health-services/health-sync-service/src/test/java/org/digit/health/sync/service/FileSyncServiceTest.java:145]: done

File: digit-health-services/health-sync-service/src/main/java/org/digit/health/sync/repository/SyncQueryBuilder.java [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+
+    public String getSQlBasedOn(SyncSearchDto syncSearchDto) {
+        StringBuilder sql = new StringBuilder("SELECT * FROM sync_log ");
```
[the remark this answers is not in the document]
Comment (jitendrac-egov) [reply digit-health-services/health-sync-service/src/main/java/org/digit/health/sync/repository/SyncQueryBuilder.java]: Updated the PR

File: digit-health-services/health-sync-service/src/test/java/org/digit/health/sync/repository/SyncRepositoryTest.java (hunk 3 above)
Comment (jitendrac-egov) [reply digit-health-services/health-sync-service/src/test/java/org/digit/health/sync/repository/SyncRepositoryTest.java]: done

File: digit-health-services/health-sync-service/src/main/java/org/digit/health/sync/repository/DefaultSyncLogRepository.java (hunk 10 above)
Comment (jitendrac-egov) [reply digit-health-services/health-sync-service/src/main/java/org/digit/health/sync/repository/DefaultSyncLogRepository.java]: updated

File: digit-health-services/health-sync-service/src/main/java/org/digit/health/sync/repository/DefaultSyncLogRepository.java (hunk 9 above)
Comment (jitendrac-egov) [reply digit-health-services/health-sync-service/src/main/java/org/digit/health/sync/repository/DefaultSyncLogRepository.java:21]: Updated

**Review by egov-joy — APPROVED** [review]

**Review by jayantp-egov — APPROVED** [review]

**Pull request merged**